core/src/main/java/io/questdb/cairo/GeoHashes.java

```diff
@@ -79,17 +79,26 @@ public static long bitmask(int count, int shift) {
         return ((1L << count) - 1) << shift;
     }
 
-    public static long fromBitString(CharSequence bits) throws NumericException {
-        if (bits.length() > MAX_BITS_LENGTH) {
-            throw NumericException.INSTANCE;
+    public static long fromBitStringNl(CharSequence bits, int start) throws NumericException {
+        int len = bits.length();
+        if (len - start <= 0) {
+            return NULL;
         }
+        return fromBitString(bits, start, Math.min(bits.length(), MAX_BITS_LENGTH));
+    }
+
+    public static long fromBitString(CharSequence bits, int start) throws NumericException {
+        return fromBitString(bits, start, Math.min(bits.length(), MAX_BITS_LENGTH));
+    }
+
+    private static long fromBitString(CharSequence bits, int start, int len) throws NumericException {
         long result = 0;
-        for (int i = 0, n = bits.length(); i < n; i++) {
+        for (int i = start; i < len; i++) {
             switch (bits.charAt(i)) {
-                case '0':
+                case 48: // '0'
```

Comment:
I like case '0': btw

Reply:
the hypothesis is that it is quicker to check the int value, that style is sprinkled over `GeoHashes` so I thought I'd be consistent.